### PR TITLE
Improve username validation

### LINKED ISSUE #975: Make validation errors for usernames descriptive
"`username` is invalid" is not really helpful, the error should state what exactly is not correct when validating (format, length, etc.)

### PR DESCRIPTION
**Pull request by pixeldesu**
Resolves #975 

Too long username:
image

Wrong format:
image

All the errors:
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -53,15 +53,20 @@ class User < ApplicationRecord # rubocop:disable Metrics/ClassLength
...
 
   before_validation do
     screen_name.strip!
   end
 
   validates :email, fake_email: true, typoed_email: true
-  validates :screen_name, presence: true, format: { with: SCREEN_NAME_REGEX }, uniqueness: { case_sensitive: false }, screen_name: true
+  validates :screen_name,
+            presence:    true,
+            format:      { with: SCREEN_NAME_REGEX },
```
Comment (nilsding) [inline app/models/user.rb]: would add a message to it too so it's not just "invalid"
```suggestion
            format:      { with: SCREEN_NAME_REGEX, message: "contains invalid characters" },
```

**Review by nilsding — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (pixeldesu) [reply app/models/user.rb]: Hmm, fair point, but in combination with the format hint right below it is kinda explaining why it is "invalid", isn't it?

**Review by raccube — APPROVED** [review]
Tested locally, happy with the logic. Agree with @nilsding's suggestion to improve the messaging. Although, please put that in a translation.

File: app/models/user.rb (hunk 1 above)
Comment (nilsding) [reply app/models/user.rb]: i'd rather have error messages be more explicit, even with the help text below.  e.g. `ü` would count as an alphanumeric character ;-)

**Review by nilsding — APPROVED** [review]

**Pull request merged**